tests/puppeteer/campaigns/tools/linkchecker.js

```diff
@@ -0,0 +1,127 @@
+require('module-alias/register');
+// Using chai
+const {expect} = require('chai');
+const helper = require('@utils/helpers');
+const files = require('@utils/files');
+const urlsList = require('./urls.js');
+
+const LOG_PASSED = process.env.LOG_PASSED || false;
+
+
+// Report dir and file
+const reportPath = 'campaigns/tools/reports';
+let filename;
+
+const output = {
+  startDate: new Date().toISOString(),
+  endDate: null,
+  pages: [],
+};
+let outputEntry = {
+  name: '',
+  url: '',
+  passed: [],
+  failed: [],
+  jsError: [],
+};
+let requestError = false;
+let requestTextError = '';
+let javascriptError = false;
+let javascriptTextError = '';
+
+let page;
+let browser;
+
+describe('Crawl every page for defects and issues', async () => {
+  before(async () => {
+    await files.createDirectory(reportPath);
+    // Create report dir
+    filename = await files.generateReportFilename();
+    browser = await helper.createBrowser();
+    page = await helper.newTab(browser);
+    await page.setExtraHTTPHeaders({
+      'Accept-Language': 'fr-FR',
+    });
+
+    await page.setRequestInterception(true);
+    // intercepts requests
+    await page.on('request', (request) => {
+      request.continue();
+    });
+    // intercepts responses
+    await page.on('response', (response) => {
+      checkResponseStatus(
+        response.request().url(),
+        response.status().toString(),
```

Comment:
```suggestion
        response.status().ok(),
```

It's maybe enough: https://github.com/puppeteer/puppeteer/blob/master/docs/api.md#responseok
Not sure the link checker needs to check for 300+